### ISSUE TITLE
ytdl-format removes metadata

### ISSUE THREAD

**Issue by sarmong** [mpv-player/mpv#11085]
### Important Information

- mpv 0.35.0
- Arch Linux 5.15
- Downloaded from Arch repo

### Reproduction steps

1. Install `yt-dlp`
2. Run `mpv --no-config https://youtube.com....`
3. Open console with `
4. Run `show-text ${metadata}`
5. Behold youtube description and other fields.
6. Run `mpv --no-config --ytdl-format=worst https://youtube.com....`
7. Repeat steps 3 and 4
8. Behold different metadata

### Expected behavior

Correct metadata for youtube video is shown regardless of `ytdl-format` option.

### Actual behavior

Youtube metadata is removed when using `ytdl-format` option

### Log file

http://0x0.st/oRO5.txt

### P.S.

Even when not using `ytdl-format`, but adding `ytdl-raw-options=embed-metadata=` I don't get the whole metadata that `yt-dlp` adds to the file when I download it directly.

**Comment by bitraid** [comment]
This has nothing to do with `ytdl-format`, but with the fact that not all formats have the same metadata. Try it with `ytdl-format=18` and `ytdl-format=134+140`, which have the same resolution/codec but the latter is served in DASH.
The `embed-metadata` option of yt-dlp, adds metadata to the file when the download is complete - it has no use with mpv.

**Comment by sarmong** [comment]
Works with `134+140` but not with `18`. Though I don't think I really understand for this means
>which have the same resolution/codec but the latter is served in DASH.

Also, initially I had this `ytdl-format=[height<=720]`, which doesn't work either.

**Comment by bitraid** [comment]
It's not that with some formats "it works" while with others it doesn't - as I previously said: **not all formats have the same metadata**.

**Comment by sarmong** [comment]
Is this metadata coming from yt-dlp?

**Comment by bitraid** [comment]
Yes, and I think you might actually be right: Some metadata are missing if the selection is not a combined format...

**Comment by bitraid** [comment]
Quick fix:
```diff
diff --git a/player/lua/ytdl_hook.lua b/player/lua/ytdl_hook.lua
index f40579ad36..5be0acf32a 100644
--- a/player/lua/ytdl_hook.lua
+++ b/player/lua/ytdl_hook.lua
@@ -516,9 +516,7 @@ local function formats_to_edl(json, formats, use_all_formats)

     -- Merge all tracks into a single virtual file, but avoid EDL if it's
     -- only a single track (i.e. redundant).
-    if #streams == 1 and single_url then
-        res.url = single_url
-    elseif #streams > 0 then
+    if #streams > 0 then
         local tags = tags_to_edl(json)
         if tags then
             -- not a stream; just for the sake of concatenating the EDL string
```

**Comment by Dudemanguy** [comment]
Should have been fixed by 8134f18c28229c38102535baf210f77d0cd6feb0.
